.upstream-tests/test/std/iterators/predef.iterators/move.iterators/move.iter.ops/move.iter.op.index/difference_type.pass.cpp

```diff
@@ -46,6 +46,9 @@ int main(int, char**)
 {
     {
         char s[] = "1234567890";
+#ifdef __NVCOMPILER
```

Comment:
Why? (I assume a compiler bug, but we should leave an explanation for this one here.)

Reply:
Yeah there is a bug. That said it is so fundamental, that it shouldnt be a compiler bug.

In a nutshell with nvc++ the array s gets initialized to char s[] = "1111111111";

Why I have absolutely no idea. Given that it works on godbolt I assume something fishy with our test infra

Reply:
We should get to the bottom of this before merging this PR then.

Reply:
That is preexisting, so I do not think it should prevent merging this

Reply:
To be clear, in the current test config the following fails on device only
```cpp
int main(int, char**)
{
    char s[] = "1234567890";
    assert(s[5] == 6);
    return 0;
}
```

I do not know whether i configured it wrong, but nvc++ does some strange things

Reply:
Okay, I didn't realize this was preexisting. We should figure this out fully, but I'm ok not blocking the PR over it.